### ISSUE TITLE
Add date filtering to the Data Cleaning tab

### ISSUE THREAD

**Issue by rumca** [akvo/akvo-flow#1160]
In the same way we can now filter reports by date range on the Export Reports tab it would be nice to add this functionality to the Data Cleaning tab as well.

**Comment by jonase** [comment]
## Test plan

Go to the data cleaning tab

* You should see date filtering options for downloading data
* Download a report without filtering
  * Check the report, no date filtering should have been applied
* Download three reports:
  * `From` date specified
  * `To` date specified
  * `From` and `to` date specified
    * Check that the downloaded reports contains the correct date ranges
* Pick one of the filtered reports and perform some data cleaning
* Import the cleaned data
* Check that the cleaned data has been updated correctly
  * Also check that the data not present in the exported/imported data (due to date filtering) **is still present and has not been altered**.

**Comment by rumca** [comment]
@jonase most of the above test plan passes without issue. I think there's a slight issue with the last point though. When I filter for a sub set of the data by date, make some alterations, upload that again, then export a fresh raw report (without dates specified) I only get the data I just cleaned and not the full set. It's still present and unaltered in the dashboard though - as I can view it on the inspect data tab.

I'm guessing here, but perhaps we need to call the cache/date invalidation for reports once we've processed the cleaned spreadsheet?

**Comment by jonase** [comment]
@rumca Was the date fields perhaps prepopulated as per https://github.com/akvo/akvo-flow/issues/1248?

**Comment by rumca** [comment]
@jonase aha! That was what was kind of what was occurring:

![screen shot 2015-04-24 at 09 22 52](https://cloud.githubusercontent.com/assets/3003031/7315294/1aef048a-ea64-11e4-9bc5-82fac8951fa8.png)

I uploaded the cleaned data, navigated to the messages tab to see if it was processed successfully, and then navigated back to the data cleaning tab to download the report again. The date filter fields were not populated, but the request still seems to use the original dates I filtered on.
